UnifiedApi/Client/Services/Submitter/Service.cs

```diff
@@ -166,11 +194,35 @@ public ServiceResult Execute(string methodName, byte[] dataArg)
         SerializedArguments = true,
       };
 
-      var taskId = SessionService.SubmitTask(dataSynapsePayload.Serialize());
+      var   taskId = "not-TaskId";
+      object[] result;
 
-      var result = ProtoSerializer.DeSerializeMessageObjectArray(SessionService.GetResult(taskId));
+      try
+      {
+        taskId = SessionService.SubmitTask(dataSynapsePayload.Serialize());
 
-      return new ServiceResult()
+        result = ProtoSerializer.DeSerializeMessageObjectArray(SessionService.GetResult(taskId));
```

Comment:
perf ?

Reply:
Even if the protoSerilizer is fast and optionel. Only if Result Type is not byte[] in the method signature